commands/command.go

```diff
@@ -262,3 +269,67 @@ func checkArgValue(v string, found bool, def Argument) error {
 func ClientError(msg string) error {
 	return &Error{Code: ErrClient, Message: msg}
 }
+
+func ExternalBinary() *Command {
+	return &Command{
+		Arguments: []Argument{
+			StringArg("args", false, true, "arguments for subcommand"),
+		},
+		External: true,
+		Run: func(req Request, res Response) {
+			binname := strings.Join(append([]string{"ipfs"}, req.Path()...), "-")
+			_, err := exec.LookPath(binname)
+			if err != nil {
+				// special case for '--help' on uninstalled binaries.
+				if req.Arguments()[0] == "--help" {
+					buf := new(bytes.Buffer)
+					fmt.Fprintf(buf, "%s is an 'external' command.\n", binname)
+					fmt.Fprintf(buf, "it does not currently appear to be installated.\n")
+					fmt.Fprintf(buf, "please refer to the ipfs documentation for instructions\n")
+					res.SetOutput(buf)
+					return
+				}
+
+				res.SetError(fmt.Errorf("%s not installed."), ErrNormal)
+				return
+			}
+
+			r, w := io.Pipe()
+
+			cmd := exec.Command(binname, req.Arguments()...)
+
+			// TODO: make commands lib be able to pass stdin through daemon
+			//cmd.Stdin = req.Stdin()
+			cmd.Stdin = io.LimitReader(nil, 0)
```

Comment:
nope. The daemon doesnt pass it through in an easy to use way, the best we get is reading it all in beforehand, and passing it as an argument.